proposals/lip-0062.md

```diff
@@ -39,35 +39,72 @@ Introduction of another hash function to the Lisk protocol was suggested in [LIP
 
 ```python
-signMessage(sk, tag, networkIdentifier, message):
-    taggedMessage = tag || networkIdentifier || message
-    hashedMessage = hash(taggedMessage)
+def signMessage(sk: bytes, tag: bytes, chainID: bytes, message: bytes) -> bytes:
+    taggedMessage = tag + chainID + message
+    hashedMessage = sha256(taggedMessage)
     return Sign(sk, hashedMessage)
-
-verifyMessageSig(pk, tag, networkIdentifier, message, signature):
-    taggedMessage = tag || networkIdentifier || message
-    hashedMessage = hash(taggedMessage)
+```
+```python
+def verifyMessageSig(pk: bytes, tag: bytes, chainID: bytes, message: bytes, signature: bytes) -> bool:
+    taggedMessage = tag + chainID + message
+    hashedMessage = sha256(taggedMessage)
     return Verify(pk, hashedMessage, signature)
 ```
 
-Where `hash` is the SHA-256 hash digest of the input, and the functions `Sign` and `Verify` are the signing and verifying functions of the respective signature scheme.
-
-### Signing and Verifying with Ed25519
-
-For the Ed25519 signature scheme, <code>[Sign](https://tools.ietf.org/html/rfc8032#section-5.1.6)</code> and <code>[Verify](https://tools.ietf.org/html/rfc8032#section-5.1.7)</code> are the signing and verifying functions as specified in [RFC 8032](https://tools.ietf.org/html/rfc8032).
+Here, <code>[Sign](https://tools.ietf.org/html/rfc8032#section-5.1.6)</code> and <code>[Verify](https://tools.ietf.org/html/rfc8032#section-5.1.7)</code> are the signing and verifying functions as specified in [RFC 8032](https://tools.ietf.org/html/rfc8032).
 
 The `signMessage` and `verifyMessageSig` functions defined in [LIP 0037](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#signing-and-verifying-with-ed25519) are superseded by the functions defined in this LIP.
 
 ### Signing and Verifying with BLS
 
-For the BLS signature scheme [used in Lisk](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0038.md#specification), `Sign` (identical to <code>[CoreSign](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-2.6)</code>) and `Verify` (identical to <code>[CoreVerify](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-2.7)</code>) are the signing and verifying functions as specified in [BLS Signatures draft-irtf-cfrg-bls-signature-04](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04).
+```python
+def signBLS(sk: bytes, tag: bytes, chainID: bytes, message: bytes) -> bytes:
+    taggedMessage = tag + chainID + message
+    hashedMessage = sha256(taggedMessage)
+    return CoreSign(sk, hashedMessage)
```

Comment:
In LIP38 it is exaplined that indeed they are the same functions (here I rather used the Core name to avoid confusion and link directly to specs), so I don't think it is needed to change LIP38. What do you mean by ' Same for CoreVerify below.'?

Reply:
I meant the same applies to `CoreVerify` and `Verify`.

Okay, then let us keep it like it is.